src/Testcontainers.DockerCompose/DockerComposeBuilder.cs

```diff
@@ -0,0 +1,147 @@
+    
+    /// <summary>   
+    /// Initializes a new instance of the <see cref="DockerComposeBuilder" /> class.
+    /// </summary>
+    public DockerComposeBuilder()
+        : this(new DockerComposeConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DockerComposeBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    private DockerComposeBuilder(DockerComposeConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override DockerComposeConfiguration DockerResourceConfiguration { get; }
+    
+    /// <inheritdoc />
+    public override DockerComposeContainer Build()
+    {
+        Validate();
+        
+        return new DockerComposeContainer(DockerResourceConfiguration, TestcontainersSettings.Logger);
+    }
+    
+    /// <summary>
+    /// Sets the compose file.
+    /// </summary>
+    /// <param name="composeFile">The compose file.</param>
+    /// <returns>A configured instance of <see cref="DockerComposeBuilder" />.</returns>
+    public DockerComposeBuilder WithComposeFile(string composeFile)
+    {
+        return Merge(DockerResourceConfiguration, new DockerComposeConfiguration
+            (composeFile: composeFile));
+    }
+    
+    /// <summary>
+    /// If true use a local Docker Compose binary instead of a container.
+    /// </summary>
+    /// <param name="localCompose">Whether the local compose will be used.</param>
+    /// <returns>A configured instance of <see cref="DockerComposeBuilder" />.</returns>
+    public DockerComposeBuilder WithLocalCompose(bool localCompose)
+    {
+        return Merge(DockerResourceConfiguration, new DockerComposeConfiguration
+            (localCompose: localCompose));
+    }
+    
+    /// <summary>
+    /// Adds options to the docker-compose command, e.g. docker-compose --compatibility.
+    /// </summary>
+    /// <param name="options">Options for the docker-compose command.</param>
+    /// <returns>A configured instance of <see cref="DockerComposeBuilder" />.</returns>
+    public DockerComposeBuilder WithOptions(params string[] options) {
+        return Merge(DockerResourceConfiguration, new DockerComposeConfiguration
+            (options: options));
+    }
+    
+    /// <summary>
+    /// Remove images after containers shutdown.
+    /// </summary>
+    /// <param name="removeImages"></param>
+    /// <returns>A configured instance of <see cref="DockerComposeBuilder" />.</returns>
+    public DockerComposeBuilder WithRemoveImages(RemoveImages removeImages) {
+        return Merge(DockerResourceConfiguration, new DockerComposeConfiguration
+            (removeImages: removeImages));
+    }
```

Comment:
Is this a Java feature?

Reply:
Yep, you can look it up [here](https://github.com/testcontainers/testcontainers-java/blob/c210afb0b0944d1a96b7a760f93d6601a9058173/core/src/main/java/org/testcontainers/containers/ComposeContainer.java#L329).